Add a few Q&As around how we work in the iOS team (#69)
Babylonpartners/ios-playbook: WorkingAtBabylon.md

<p align="center">
<img src="logo.png">
</p>

Working in the iOS team
==================================

TODO

> This is more or less modelled after https://github.com/codeOfRobin/Questions-to-ask-your-interviewer-iOS

Author: Please rewrite nicely.

### Where do we store our code?
We use Github. We have a central repo that hosts the main and white label apps, and a few Open Source projects that we created and rely on.

## Continuous Integration
### How many apps/targets/schemes?
### How big is your Fastfile?
It's around 250 lines of code at the moment, with about 15-20 lanes for beta testing, daily and custom builds, etc.

### Does it have several environment references?
### Can you dry run a lane on your machine right now?

> Yes.

Author: Which ones are not runnable?

### Codesigning?
  
* What architectural patterns are you using to make development easier or more consistent? What problems with your current approach have you come across after settling on one (if you have one)?

## Team/People
### comms - slack?
Slack, as a non-trivial part of the team is remote. Zoom for meetings of more than 2-3 people.

### Do people ever take sick days to get work done?
No.

### When was the last time someone worked a late night/weekend in order to meet a deadline?
TODO

### How many meetings do you have in a day?

> It depend on the squad you’re assigned. For Enrolment and Integrity I counted 18 in a typical sprint, so that averages to ~2 a day. Note that quite a few of them are optional.

Author: ```suggestion
It depend on the squad you’re assigned. For Enrolment and Integrity I counted 18 in a typical sprint, so that averages to ~2 a day. Note that quite a few of them are optional.

Interviews, sometimes, can take up some time. It's usually, at maximum, one hour per week.
```

> It depend on the squad you’re assigned. For Enrolment and Integrity I counted 18 in a typical sprint, so that averages to ~2 a day. Note that quite a few of them are optional.

Author: Added.

### Does your calendar look like a losing game of tetris? Do you have large continuous uninterrupted hours?

> Most of the meetings are on the first Monday/Tuesday of the sprint.

Reviewer: This is totally squad dependent. In Triage UI for example, this is not the case. It's meetings all over the place.

Reviewer 2: Same i'm afraid

Author: ```suggestion
It depends a lot on the squad. For Enrolment and Integrity, for example, most meetings are on the first Monday and Tuesday of the sprint. For Triage and Health Management this is not the case and are more spread out.
```

### Are there any private spaces in your office?
There are very few at the moment.

### Is there a gym/field/court close to your office for physical activity?
There is a commercial gym close by, a park, and a barebones gym in the office. There’s also free yoga twice a week.

### Do people feel comfortable calling attention to systemic issues/heavily “band-aid”-ed systems?

> Yes, we have anonymous company level Q&A and peakon surveys. [this is my experience].

Reviewer: The company level QA will no longer be anonymous, if what was said yesterday in the company standup is true.

### Are any of your engineering managers mobile engineers? Do mobile devs feel like they have to learn backend dev to "prove themselves" / feel like they have a seat at the table?
iOS engineers have to know iOS, and that’s it.

## Designers?
### What’s the handoff process like?
As we work in squads, collaboration with designers, product, and QA is very tight and happens every day. After a ticket has been discussed in discovery and planning, designs are provided on Zeplin (sketch).

### Do you let devs get down and dirty with sketch files?

> Not usually, but developers are encouraged to report wrong alignments, spacing, and in general to partecipate in the design/UX discussion.

Reviewer 3: ```suggestion
Not usually, but developers are encouraged to report wrong alignments, spacing, and in general to participate in the design/UX discussion.
```

> ### Do you focus on making your android and iOS apps look consistent?

Reviewer 3: ```suggestion
### Do you focus on making your Android and iOS apps look consistent?
```

There is a drive for consistency with a design system, but platform-specific conventions are taken into account.

> ### Example of the above: Do you have the “title” of a VC on the left in iOS/on the center in android for consistency? Do you have a modal View Controller presented with a back arrow on the left?

Reviewer 3: ```suggestion
### Example of the above: Do you have the “title” of a VC on the left in iOS/on the center in Android for consistency? Do you have a modal View Controller presented with a back arrow on the left?
```

We follow iOS standards, but it's up to every squad.

> ## Q/A?

Reviewer 3: ```suggestion
## QA?
```

### What’s the handoff-to-QA process like?
Same answer as with designers. Collaboration is frequent as there is a QA per squad. A build is triggered from the main branch every night (with custom builds trigger-able at any time), and in the morning the QA starts testing against it.

## Learning
### Do you sponsor conference tickets?
We have a set `training` budget that can be used at the person's discretion.

### Books/online resources?
We have a set `training` budget that can be used at the person's discretion.

## Product
### Do devs get access to support tickets?
### How involved are devs in product decisions?
There is usually a product person in each squad that leads this, but everyone is encouraged to contribute and/or raise concerns.

### Do you consider the societal impact of the products you’re making?
Yes. We wouldn't be here otherwise.

> ### Does your marketing org ensure your marketing materials are inclusive (I like to call this the Revolut filter)?

Reviewer 3: I don't get the point of this. Sounds like some accusation of a third company, the document should be free of such things imo.

Author: Will delete.

> ### Does your marketing org ensure your marketing materials are inclusive (I like to call this the Revolut filter)?

Author: ```suggestion
```

<p align="center">
<img src="logo.png">
</p>

Working in the iOS team
==================================

> 

Author: ```suggestion
```

> TODO

Author: ```suggestion
```

> This is more or less modelled after https://github.com/codeOfRobin/Questions-to-ask-your-interviewer-iOS

Author: ```suggestion
This is document is modelled after Robin Malhotra's [Questions to ask your iOS interviewer][https://github.com/codeOfRobin/Questions-to-ask-your-interviewer-iOS). It aims to provide a few answers to how we work at Babylon and our process is structured.
```

## Continuous Integration
### How many apps/targets/schemes?
### How big is your Fastfile?
It's around 250 lines of code at the moment, with about 15-20 lanes for beta testing, daily and custom builds, etc.

### Does it have several environment references?
### Can you dry run a lane on your machine right now?
Yes.

### Codesigning?
  
* What architectural patterns are you using to make development easier or more consistent? What problems with your current approach have you come across after settling on one (if you have one)?

## Team/People
### comms - slack?
Slack, as a non-trivial part of the team is remote. Zoom for meetings of more than 2-3 people.

### Do people ever take sick days to get work done?
No.

### When was the last time someone worked a late night/weekend in order to meet a deadline?
TODO

### How many meetings do you have in a day?
It depend on the squad you’re assigned. For Enrolment and Integrity I counted 18 in a typical sprint, so that averages to ~2 a day. Note that quite a few of them are optional.

> Interviews, sometimes, can take up some time. It's usually, at maximum, one hour per week.

Reviewer 2: Should we state that we also need to analyze demo projects? While not meeting related per se it is related with the interview/hiring proccesses as well

### Does your calendar look like a losing game of tetris? Do you have large continuous uninterrupted hours?
It depends a lot on the squad. For Enrolment and Integrity, for example, most meetings are on the first Monday and Tuesday of the sprint. For Triage and Health Management this is not the case and are more spread out.

### Are there any private spaces in your office?
There are very few at the moment.

### Is there a gym/field/court close to your office for physical activity?

> There is a commercial gym close by, a park, and a barebones gym in the office. There’s also free yoga twice a week and a football club

Reviewer 2: Barebones gym only in Babylon West, right?

### Do people feel comfortable calling attention to systemic issues/heavily “band-aid”-ed systems?
Yes, we have anonymous company level Q&A and peakon surveys. [this is my experience].

### Are any of your engineering managers mobile engineers? Do mobile devs feel like they have to learn backend dev to "prove themselves" / feel like they have a seat at the table?

> iOS engineers have to know iOS and general programming practices, and that’s it.

Reviewer 2: The first part of the question is not addressed and regarding the second bit I think we should say that every engineer can and should make proposals to improve our processes (thus, everyone has a seat at the table).

Reviewer 4: Yep. Worth mentioning that are current line manager and the level above him are both (or were at some point) heavily involved in mobile development

<p align="center">
<img src="logo.png">
</p>

Working in the iOS team
==================================

This document is modelled after Robin Malhotra's [Questions to ask your iOS interviewer](https://github.com/codeOfRobin/Questions-to-ask-your-interviewer-iOS). It aims to provide a few answers to how we work at Babylon and our process is structured.

### Where do we store our code?
We use Github. We have a central repo that hosts the main and white label apps, and a [few Open Source projects](https://github.com/search?q=topic%3Aios+org%3ABabylonpartners+fork%3Atrue) that we created and rely on.

## Continuous Integration
### What Continuous Integration service do you use?

We currently run on CircleCI

### How many apps/targets/schemes?

> We have a main app and a few white-label apps.

Reviewer 5: ~~a main~~ -> *the* main

> We have a main app and a few white-label apps.

Author: ```suggestion
We have the main app and a few white-label apps.
```

We use a modular architecture, so our app workspace is split into 9 Xcode projects for modules and apps, 3 projects for test utilities and shared features, and 4 projects related to our SDK, not counting the Pods project (and our OSS projects)

> Each module project has 3 targets on average: one for the framework, one for the Unit Test and one for Snapshot tests when applicable. The app project has 18 targets (6*3) given our white-labelling variants.

Reviewer 4: >one for the framework, one for the Unit Test and one for Snapshot tests

Usually the snapshot tests and typical unit tests are part of the same framework, no?

Reviewer 4: Looks like BabylonUI is the only project with three frameworks

### How big is your Fastfile?
It's around 250 lines of code at the moment, with about 15-20 lanes for beta testing, daily and custom builds, etc.

### Does it have several environment references?
### Can you dry run a lane on your machine right now?
Yes.

### Codesigning?
  
* What architectural patterns are you using to make development easier or more consistent? What problems with your current approach have you come across after settling on one (if you have one)?

## Team/People
### comms - slack?
Slack, as a non-trivial part of the team is remote. Zoom for meetings of more than 2-3 people.

### Do people ever take sick days to get work done?
No.

### When was the last time someone worked a late night/weekend in order to meet a deadline?
TODO

### How many meetings do you have in a day?

> It depend on the squad you’re assigned. For Enrolment and Integrity I counted 18 in a typical sprint, so that averages to ~2 a day. Note that quite a few of them are optional.

Reviewer 5: ~~it depend~~ -> it *depends*

> Interviews, sometimes, can take up some time. It's usually, at maximum, one hour per week.

Reviewer 4: Could be worth mentioning that they are rotated around the team equally, and that developers of all levels are involved. I believe most companies do not include juniors on this

> Interviews, sometimes, can take up some time. It's usually, at maximum, one hour per week.

Author: ```suggestion
Interviews, sometimes, can take up some time. It's usually, at maximum, one hour per week. Reviewing tests is also part of your day-to-day duties.
```

### Does your calendar look like a losing game of tetris? Do you have large continuous uninterrupted hours?
It depends a lot on the squad. For Enrolment and Integrity, for example, most meetings are on the first Monday and Tuesday of the sprint. For Triage and Health Management this is not the case and are more spread out.

### Are there any private spaces in your office?
There are very few at the moment.

### Is there a gym/field/court close to your office for physical activity?

> There is a commercial gym close by, a park, and a barebones gym in the office. There’s also free yoga twice a week and a football club

Reviewer 4: ```suggestion
There is a commercial gym close by, a park, and a barebones gym in the office. There’s also free yoga twice a week and a weekly 5-a-side football game.
```
We don't play against other companies (yet) so I'm not sure we could call it a "club" :p
